packages/suite/src/components/firmware/FirmwareInitial.tsx

```diff
@@ -8,28 +10,64 @@ import {
 import { Translation } from '@suite-components';
 import { useDevice, useFirmware, useOnboarding } from '@suite-hooks';
 import { ReconnectDevicePrompt, InstallButton, FirmwareOffer } from '@firmware-components';
-import { TrezorDevice } from '@suite/types/suite';
-import { getFwVersion, getFwUpdateVersion } from '@suite-utils/device';
+import { FirmwareType, TrezorDevice } from '@suite-types';
+import { getFwVersion, getFwUpdateVersion, isBitcoinOnly } from '@suite-utils/device';
+
+const InfoRow = styled.div`
+    align-items: center;
+    display: flex;
+    font-weight: 600;
+    gap: 4px;
+    margin: 8px auto 0 auto;
+    max-width: max-content;
+`;
+
+const ButtonRow = styled.div`
+    display: flex;
+    gap: 20px;
+`;
+
+const TextButton = styled.button`
+    background: none;
+    border: none;
+    cursor: pointer;
+    font-size: inherit;
+    padding: 0;
+    text-decoration: underline;
+`;
+
+const StyledButton = styled(Button)`
+    min-width: 180px;
+`;
 
 interface FirmwareInitialProps {
     cachedDevice?: TrezorDevice;
     setCachedDevice: React.Dispatch<React.SetStateAction<TrezorDevice | undefined>>;
     // This component is shared between Onboarding flow and standalone fw update modal with few minor UI changes
     // If it is set to true, then you know it is being rendered in standalone fw update modal
     standaloneFwUpdate?: boolean;
-    onInstall: () => void;
+    onInstall: (firmwareType?: FirmwareType) => void;
+    switchType?: boolean;
 }
 
 const getDescription = ({
     required,
     standaloneFwUpdate,
     reinstall,
+    switchType,
```

Comment:
I am a bit confused by that naming, not only in this file. Also some code that follows like:

```
const targetType =
        isBitcoinOnly(device) === !!switchType ? FirmwareType.Universal : FirmwareType.BitcoinOnly;
```

is also unclear because of that 😅 Should it indicate a possibility to switch types or that the type has been switched?

Reply:
I made the names and this line more explicit in 427f67466f36b33deb3c033751dfce922a5e3f00